### DISCUSSION TITLE
Dynamic Redirects ctx is undefined

### DISCUSSION THREAD

**Discussion by julydevelops** [gladly-team/next-firebase-auth#339] [Q&A]
I'm trying to capture the `ctx.req` but the `ctx` is coming in `undefined` 

Does anyone have thoughts as to why this is happening?

```
authPageURL: ({ ctx }) => {
      const { req, resolvedUrl } = ctx // ctx undefined here
  
      // Custom logic

      return '/redirect-path'
    }
```

**Comment by kmjennison** [comment]
The `ctx` value will only be available server-side. On the client side, you can rely on `window`. Here's how the example app handles it:
https://github.com/gladly-team/next-firebase-auth/blob/main/example/utils/initAuth.js#L15

Duplicate of https://github.com/gladly-team/next-firebase-auth/issues/311.
